### ISSUE TITLE
Issue with contact forces when computing r damage in strike task

### ISSUE THREAD

**Issue by jacktheripper19** [inspirai/TimeChamber#7]
Hello,

When trying to see the value of each reward term for the task of striking, I noticed that the r_damage can be triggered even when the agents are not hitting each other. After further investigation, I found that the contact buffers can also include self collision in addition to contact between different agents; for example, if an agent's hand collide with its own head it will be included in the contact buffer, therefore it will be used in the reward as if it got hit by the opponent which was not the case.

This could be solved by filtering the contact buffer by specifying the bodies involved in the contact, but I saw that this functionality is not available right now in isaacgym. The workaround I can think of is to detect first if there was contact between specific bodies by measuring the distance between them, once the distance is lower than some threshold, you assume there was contact, then you check the magnitude of the contact force.

Hope I was clear.

**Comment by ZeldaHuang** [comment]
@jacktheripper19 you are right,your solution will work in most situations (when the distance is lower than threshold, self damage will still be included in r_damage), feel free to create pr.
